pagopa/io-app
# fix: [IOCOM-1091] Fix for duplicate message in messages list

## Short description
This PR fixes a condition where a new message could appear twice in the messages list.

## List of changes proposed in this pull request
In the Message List component:
- Proper use of focus effect, using React's `useFocusEffect` with proper dependencies
- Loading of previous messages page (with focus effect) does not trigger anymore when the component is already refreshing all messages. 

Latter is what was causing the bug: two subsequent asynchronous requests (one for all messages and one for just the new one - the previous page) where the first retrieved all messages while the second just the new ones, using the current-but-soon-to-be-outdated starting index (`minimum_id`). 
So, the reload-all request ends, filling the redux state with all new and old messages. If the previous-page request ends after that, its data has some duplicates in it (the new messages - since its starting `minimum_id` was the old one, not the current one received from the last reload-all) but the reducer puts them in front of the previously receveid ones from the reload-all, duplicating some of them.

## How to test
You can use the io-dev-api-server code [on this PR](https://github.com/pagopa/io-dev-api-server/pull/349) to create a new message.

Do the following in less than 60 seconds (the refresh time): select a message, create a new one (`POST message` endpoint) and go back to the message list. Now wait for the 60 seconds to expire and use the pull-down-to-refresh. On the `main` branch, two duplicated messages should appear on top of the list. On this branch, there is no duplicate.
